src/goto-checker/goto_checker.h

```diff
@@ -0,0 +1,70 @@
+/*******************************************************************\
+
+Module: Goto Checker Interface
+
+Author: Daniel Kroening, Peter Schrammel
+
+\*******************************************************************/
+
+/// \file
+/// Goto Checker Interface
+
+#ifndef CPROVER_GOTO_CHECKER_GOTO_CHECKER_H
+#define CPROVER_GOTO_CHECKER_GOTO_CHECKER_H
+
+#include <goto-programs/goto_trace.h>
+
+#include <util/options.h>
+#include <util/ui_message.h>
```

Comment:
My preference is to use forward declarations (of `optionst` and `ui_message_handlert`) and just include `message.h` to reduce build dependencies.

Reply:
Not for class members, though.